contracts/vaults/TreasuryVault/RibbonAutocallVault.sol

```diff
@@ -0,0 +1,519 @@
+
+        nObsFreq = _obsFreq;
+        nPeriod = _period;
+    }
+
+    /**
+     * @dev Overrides RibbonTreasuryVault commitAndClose()
+     */
+    function commitAndClose() external override nonReentrant {
+        address currentOption = optionState.currentOption;
+
+        IOtoken currentOToken = IOtoken(currentOption);
+        uint256 expiry =
+            currentOption == address(0) ? 0 : currentOToken.expiryTimestamp();
+        uint256 strikePrice =
+            currentOption == address(0) ? 0 : currentOToken.strikePrice();
+
+        
+        (uint256 autocallTS, uint256 nCBBreaches, uint256 lastCBBreach) =
+            _autocallState(expiry);
+
+        // Calculate coupons earned
+        (, uint256 earnedAmt, uint256 returnAmt) =
+            _couponsEarned(nCBBreaches, lastCBBreach);
+
+        // If before expiry, attempt to autocall
+        if (block.timestamp < expiry) {
+            // Require autocall barrier hit at least once
+            require(autocallTS < block.timestamp, "R10");
+            // Burn the unexpired oTokens
+            _burnRemainingOTokens();
+            // Require vault possessed all oTokens sold to counterparties
+            require(vaultState.lockedAmount == 0, "R11");
+        }
+
+        if (earnedAmt > 0) {
+            // Transfer earned coupons to autocall buyer
+            transferAsset(autocallBuyer, earnedAmt);
+        }
+
+        if (returnAmt > 0) {
+            // Transfer unearned coupons back to autocall seller
+            transferAsset(autocallSeller, returnAmt);
+        }
+
+        // Commit and close vanilla put
+        super._commitAndClose();
+
+        // Commit and close enhanced put
+        _commitAndCloseEnhancedPut(expiry, strikePrice);
+
+        // Set coupon state
+        CouponState memory _couponState = couponState;
+        couponState.couponType = _couponState.nCouponType;
+        couponState.AB = _couponState.nAB;
+        couponState.CB = _couponState.nCB;
+
+        // Set observation period frequency
+        obsFreq = nObsFreq;
+        period = nPeriod;
+        numTotalObs = (period * 1 days) / obsFreq;
+    }
+
+    /**
+     * @dev Settles the enhanced put
+     * @param _expiry is the expiry of the current option
+     * @param _oldStrikePrice is the strike of the current option
+     */
+    function _commitAndCloseEnhancedPut(
+        uint256 _expiry,
+        uint256 _oldStrikePrice
+    ) internal {
+        uint256 expiryPrice =
+            ORACLE.getExpiryPrice(vaultParams.underlying, _expiry);
+
+        PutOption memory _putOption = putOption;
+
+        // If put ITM, transfer to autocall seller
+        if (
+            _putOption.payoff > 0 &&
+            expiryPrice <= _oldStrikePrice &&
+            block.timestamp > _expiry
+        ) {
+            // Transfer current digital option payoff
+            transferAsset(
+                autocallSeller,
+                (oTokenMintAmount * _putOption.payoff) / 10**8
```

Comment:
`10**8` could be `10**Vault.OTOKEN_DECIMALS`

Reply:
yes, lets change it